### PR TITLE
brief editorial + c++ solution to queue composite

### PR DESCRIPTION
**Pull request by ryanchou-dev**
digging through my old code :p

---
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/gold/ys-QueueComposite.mdx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+id: ys-QueueComposite
+source: YS
+title: Queue Composite
+author: Ryan Chou
+---
+
+
+## Explanation
+
+Note that we can compute the portion with $x$ as a factor independently from the constant in the equation. Thus, we can support $\mathcal{O}(\log n)$ removals, and $\mathcal{O}(1)$ additions and queries.
```
Comment (SansPapyrus683) [inline solutions/gold/ys-QueueComposite.mdx:11]: is this...really all there is?

Comment (ryanchou-dev) [reply solutions/gold/ys-QueueComposite.mdx]: that was my thought process 😳 

is there a different intended solution? i couldn't find any other solutions online...

Comment (SansPapyrus683) [reply solutions/gold/ys-QueueComposite.mdx]: seems like one of those one-line editorials where the reader ends up having more questions than answers

Comment (ryanchou-dev) [reply solutions/gold/ys-QueueComposite.mdx]: word

File: solutions/gold/ys-QueueComposite.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+int main() {
+	int q;
+	cin >> q;
+
+	ll mult_x = 1;
+	ll constant = 0;
+
+	queue<pair<int, int>> f;
+
+	for (int _ = 0; _ < q; _++) {
```
Comment (SansPapyrus683) [inline solutions/gold/ys-QueueComposite.mdx:67]: replace w/ `i` or smth (c++ doesn't do this underscore stuff)

File: solutions/gold/ys-QueueComposite.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+int main() {
+	int q;
+	cin >> q;
+
+	ll mult_x = 1;
+	ll constant = 0;
+
+	queue<pair<int, int>> f;
+
```
Comment (SansPapyrus683) [inline solutions/gold/ys-QueueComposite.mdx:66]:
```suggestion
```

File: solutions/gold/ys-QueueComposite.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+const ll MOD = 998244353;
+
+ll bin_exp(ll x, ll n) {
```
Comment (SansPapyrus683) [inline solutions/gold/ys-QueueComposite.mdx:44]: link to guide w/ sauce

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
yeah this is good (just like all your other prs ryan)

File: solutions/gold/ys-QueueComposite.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/gold/ys-QueueComposite.mdx:66]: the more you learn

File: solutions/gold/ys-QueueComposite.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/gold/ys-QueueComposite.mdx:45]: okayy

File: solutions/gold/ys-QueueComposite.mdx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+			f.pop();
+
+			// compute modular inverse with exponentation
+			ll inv = bin_exp(x.first, MOD - 2);
+
+			mult_x = (mult_x * inv) % MOD;
+			constant -= (x.second * mult_x) % MOD;
+
+			if (constant < 0) { constant += MOD; }
+		} else {
```
Comment (SansPapyrus683) [inline solutions/gold/ys-QueueComposite.mdx:96]: make this an `else if` as well

**Review by SansPapyrus683 — APPROVED** [review]
alr

**Review by nwatx — APPROVED** [review]

**Pull request merged**